### ISSUE TITLE
--strict does not complain when non-existent markers are specified as args to pytest

### ISSUE THREAD

**Issue by geokala** [pytest-dev/pytest#2781]
I'm not seeing pytest complain in strict mode when filtering by markers that I haven't specified.
e.g.  pytest --strict -m marker_that_i_tyoped_wrong mytests/
This will complete quite happily, deselecting all tests.

It would be nice if the behaviour were able to be changed, perhaps by another flag (to avoid causing sadness to anyone relying on the current behaviour), such that if the mark expression contained invalid marks it would cause a failure in the same way that a test containing invalid marks would.

**Comment by RonnyPfannschmidt** [comment]
we do support matching partial marks,
in order to correctly support this, we'd have to report a warning after collection and it might be wrong when passing a different set files

so in general, this one is actually tricky to solve

**Comment by geokala** [comment]
Yeah, I guessed it was probably somewhat unpleasant or it would've already been there.

As an idea, would something like --strict-markexpr disabling partial mark matching and then complaining on any undefined marks in the markexpr be viable, or would it still have the same problem you stated?

**Comment by RonnyPfannschmidt** [comment]
thats not clear to me, it would need some experimentation

as the marker and keyword mechanism is changing, it may be a good idea to think about a more consistent way to query for tests that doesn't have to share the restriction of mark expressions or keyword expressions
